### PR TITLE
datavalues2: conditional function: if

### LINKED ISSUE #3969: datavalues2 && conditional functions

### PR DESCRIPTION
**Pull request by Veeupup**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

* finish conditional function `if`
* make `ColumnViewer` work with `ConstColumn`

## Changelog

- New Feature

## Related Issues

Fixes #3969 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by PsiACE** [comment]
@mergify update

File: common/functions/src/scalars/conditionals/if.rs [hunk 1]
```diff
@@ -11,59 +11,144 @@
...
+                    Ok(builder.build(size))
+                }
+            }};
+        }
+
+        with_match_primitive_type2!(type_id, |$T| {
+            primitive_build!($T)
+        }, {
+            match type_id {
+                TypeID::String => primitive_build!(Vu8),
```
Comment (sundy-li) [inline common/functions/src/scalars/conditionals/if.rs:145]: with_match_primitive_type2 

=> with_match_scalar_types2  to cover these two?

Comment (Veeupup) [reply common/functions/src/scalars/conditionals/if.rs]: yes, resolved

File: common/functions/src/macros.rs [hunk 2]
```diff
@@ -11,3 +11,34 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+#[macro_export]
+macro_rules! with_match_primitive_type2 {
```
Comment (sundy-li) [inline common/functions/src/macros.rs:16]: rename primitive to be scalar?

Comment (Veeupup) [reply common/functions/src/macros.rs]: got it.

**Review by sundy-li — APPROVED** [review]

**Review by PsiACE — APPROVED** [review]

File: common/functions/src/scalars/conditionals/if.rs [hunk 3]
```diff
@@ -11,59 +11,138 @@
...
-        // The condition of 'Null' is treated as false. SELECT IF(NULL, "YES", "NO") -> "No"
-        let nullable = args[1].is_nullable() || args[2].is_nullable();
+    fn return_type(&self, args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        let l_dt = remove_nullable(args[1]);
+        let r_dt = remove_nullable(args[2]);
 
-        let mut aggregate_args = Vec::with_capacity(args.len() - 1);
-        for arg in args.iter().skip(1) {
-            aggregate_args.push(arg.data_type().clone())
+        let mut least_supertype = compare_coercion(&l_dt, &r_dt)?;
```
Comment (zhyass) [inline common/functions/src/scalars/conditionals/if.rs:55]: Here need use `aggregate_types` instead of `compare_coercion`

Comment (Veeupup) [reply common/functions/src/scalars/conditionals/if.rs:55]: thanks for reminding,, `aggregate_types` is better, I will make another pr to make it.

**Pull request merged**